yarn-project/simulator/src/public/fixtures/opcode_spammer.ts

```diff
@@ -0,0 +1,1081 @@
+ */
+function createSetInstruction(offset: number, memValue: MemoryValue): Bufferable {
+  const tag = memValue.getTag();
+  const value = memValue.toBigInt();
+
+  // SET_8 only supports offset <= 255 and value <= 255
+  if (offset <= 0xff && value <= 0xffn) {
+    return new Set(0, offset, tag, Number(value)).as(Opcode.SET_8, Set.wireFormat8);
+  }
+  // SET_16+ support offset <= 65535
+  if (value <= 0xffffn) {
+    return new Set(0, offset, tag, Number(value)).as(Opcode.SET_16, Set.wireFormat16);
+  }
+  if (value <= 0xffffffffn) {
+    return new Set(0, offset, tag, Number(value)).as(Opcode.SET_32, Set.wireFormat32);
+  }
+  if (value <= 0xffffffffffffffffn) {
+    return new Set(0, offset, tag, value).as(Opcode.SET_64, Set.wireFormat64);
+  }
+  if (value <= 0xffffffffffffffffffffffffffffffffn) {
+    return new Set(0, offset, tag, value).as(Opcode.SET_128, Set.wireFormat128);
+  }
+  return new Set(0, offset, tag, value).as(Opcode.SET_FF, Set.wireFormatFF);
+}
+
+/**
+ * Append (to the instructions array) the SET instructions for the setup.
+ *
+ * @param instructions - the instructions array to append the setup to
+ * @param setup - the setup configuration specifying what SETs to do
+ */
+function appendSetupInstructions(instructions: Bufferable[], setup: SetupItem[]): void {
+  for (const item of setup) {
+    if (typeof item === 'function') {
+      // item is a function that creates setup instructions (like)
+      instructions.push(...item());
+    } else {
+      // MemSetup
+      instructions.push(createSetInstruction(item.offset, item.value));
+    }
+  }
+}
+/**
+ * Append (to the instructions array) the target instructions nTimes times.
+ *
+ * @param instructions - the instructions array to append the loop to
+ * @param config - the spam config to use
+ * @param nTimes - the number of times to append the target instructions
+ * @returns the number of target instructions appended
+ */
+function appendTargetNTimes(instructions: Bufferable[], config: SpamConfig, nTimes: number) {
+  for (let i = 0; i < nTimes; i++) {
+    instructions.push(...config.targetInstructions());
+  }
+}
+
+/**
+ * Append (to the instructions array) an infinite loop that maximizes target instruction density.
+ * Fills remaining bytecode space with unrolled target instructions.
+ *
+ * @param instructions - the instructions array to append the loop to
+ * @param config - the spam config to use
+ * @returns the number of target instructions in the loop body
+ */
+function appendInfiniteLoop(instructions: Bufferable[], config: SpamConfig): number {
+  const setupBytecode = encodeToBytecode(instructions);
+  const setupSize = setupBytecode.length;
+
+  // Compute the size of the target instruction(s)
+  const targetSize = encodeToBytecode(config.targetInstructions()).length;
+
+  // Fill remaining space (loop body) with target instructions
+  const availableForLoopBody = MAX_BYTECODE_BYTES - setupSize - JUMP_SIZE;
```

Comment:
I wonder if we _need_ to use all the bytes. IIUC this would be _good_, because in simulation it doesnt matter, and in proving, you not only put pressure on the instructions, but also on hashing.